[R2] Add upload/download examples to rclone example (#5168)
cloudflare/cloudflare-docs: content/r2/examples/rclone.md

You may then use the new `rclone` provider for any of your normal workflows.

> ## Listing buckets & objects

Reviewer: ```suggestion
## List buckets & objects
```

# └── todos.txt
```

> ## Uploading & retrieving objects

Reviewer: ```suggestion
## Upload and retrieving objects
```

> The [rclone copy](https://rclone.org/commands/rclone_copy/) command can be used to upload objects to an R2 bucket and vice versa - this allows you to upload files up to the 5TB maximum object size that R2 supports.

Reviewer: ```suggestion
The [rclone copy](https://rclone.org/commands/rclone_copy/) command can be used to upload objects to an R2 bucket and vice versa - this allows you to upload files up to the 5 TB maximum object size that R2 supports.
```

  
```sh
# Upload dog.txt to the user-uploads bucket
$ rclone copy dog.txt r2demo:user-uploads/dog.txt
$ rclone tree r2demo:user-uploads
# /
# ├── foobar.png
# └── dog.txt

# Download dog.txt from the user-uploads bucket
$ rclone copy r2demo:user-uploads/dog.txt dog.txt
```

> ## Generating presigned URLs

Reviewer: ```suggestion
## Generate presigned URLs
```

# └── todos.txt
```

> ## Upload and retrieving objects

Reviewer: ```suggestion
## Upload and retrieve objects
```
